### PR TITLE
fix: input engine and logs tests

### PR DESCRIPTION
**Pull request by pdepaepe**
Hello,

This PR will handle the new input engine endpoint of dbaas/logs api and also fix various tests.

Thanks for your time on this.

Signed-off-by: Pierre De Paepe <pierre.de-paepe@ovhcloud.com>

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-check#1201.0 ✔ 
* 
  * checks ✔

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-check#1200.0 ✔ 
* 
  * checks ✔

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-testacc#1200.0 ✘  
* 
  * checks ✘

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-check#1204.0 ✔ 
* 
  * checks ✔

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-check#1203.0 ✔ 
* 
  * checks ✔

**Review by yanndegat — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-testacc#1204.0 ✘  
* 
  * checks ✘

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-testacc#1203.0 ✘  
* 
  * checks ✘

File: ovh/data_dbaas_logs_input_engine_test.go [hunk 1]
```diff
@@ -1,33 +1,49 @@
...
- name          = "logstash"
- version       = "7.x"
+ service_name  = "%s"
+ name          = "%s"
+ version       = "%s"
 }
 `
 
 func TestAccDataSourceDbaasLogsInputEngine_basic(t *testing.T) {
+	serviceName := os.Getenv("OVH_DBAAS_LOGS_SERVICE_TEST")
```
Comment (scraly) [inline ovh/data_dbaas_logs_input_engine_test.go:29]: Is it possible to add a precheck on this new environment variable `OVH_DBAAS_LOGS_SERVICE_TEST` please?
in order to skip the test and not fall in error if the env variable is not filled:

```
$ make testacc TESTARGS="-run TestAccDataSourceDbaasLogsInputEngine_deprecated"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run TestAccDataSourceDbaasLogsInputEngine_deprecated -timeout 240m
?   	github.com/ovh/terraform-provider-ovh	[no test files]
=== RUN   TestAccDataSourceDbaasLogsInputEngine_deprecated
    data_dbaas_logs_input_engine_test.go:77: Step 1/1 error: Error running pre-apply refresh: exit status 1

        Error: Error calling GET /dbaas/logs//input/engine/xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx:
        	 "HTTP Error 500: Server::InternalServerError: \"Internal server error\" (X-OVH-Query-Id: EU.ext-5.639b1a05.xxxx.xxxxx-xxxx-xxxx-xxxx-xxxxxxx)"

          with data.ovh_dbaas_logs_input_engine.logstash,
          on terraform_plugin_test.tf line 2, in data "ovh_dbaas_logs_input_engine" "logstash":
           2: data "ovh_dbaas_logs_input_engine" "logstash" {

--- FAIL: TestAccDataSourceDbaasLogsInputEngine_deprecated (3.94s)
FAIL
FAIL	github.com/ovh/terraform-provider-ovh/ovh	4.140s
?   	github.com/ovh/terraform-provider-ovh/ovh/helpers	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/ovh/terraform-provider-ovh/ovh/helpers/hashcode	(cached) [no tests to run]
FAIL
make: *** [testacc] Error 1
```

And add this ne env variable in the homepage please

**Review by scraly — CHANGES_REQUESTED** [review]

**Review by scraly — CHANGES_REQUESTED** [review]

File: ovh/data_dbaas_logs_input_engine_test.go [hunk 2]
```diff
@@ -1,33 +1,49 @@
...
+
+	config := fmt.Sprintf(
+		testAccDataSourceDbaasLogsInputEngine_basic,
+		serviceName,
+		name,
+		version,
+	)
+
 	resource.Test(t, resource.TestCase{
 		PreCheck: func() { testAccPreCheckCredentials(t) },
```
[the remark this answers is not in the document]
Comment (pdepaepe) [reply ovh/data_dbaas_logs_input_engine_test.go]: Fixed, thanks

File: ovh/data_dbaas_logs_input_engine_test.go (hunk 1 above)
Comment (pdepaepe) [reply ovh/data_dbaas_logs_input_engine_test.go:29]: Should be better now

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-check#1205.0 ✔ 
* 
  * checks ✔

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-testacc#1205.0 ✘  
* 
  * checks ✘

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-testacc#1206.0 ✘  
* 
  * checks ✘

File: website/docs/d/dbaas_logs_input_engine.html.markdown [hunk 3]
```diff
@@ -14,6 +14,7 @@ Use this data source to retrieve information about a DBaas logs input engine.
 
 ```hcl
 data "ovh_dbaas_logs_input_engine" "logstash" {
+  service_name = "XXXXXX"
```
Comment (scraly) [inline website/docs/d/dbaas_logs_input_engine.html.markdown:17]:
```suggestion
  service_name = "ldp-xx-xxxxx"
```

File: website/docs/d/dbaas_logs_input_engine.html.markdown [hunk 4]
```diff
@@ -22,10 +23,11 @@ data "ovh_dbaas_logs_input_engine" "logstash" {
 
 ## Argument Reference
 
+* `service_name` - The service name
```
Comment (scraly) [inline website/docs/d/dbaas_logs_input_engine.html.markdown:26]:
```suggestion
* `service_name` - The service name. It's the ID of your Logs Data Platform instance.
```

**Review by scraly — CHANGES_REQUESTED** [review]

File: website/docs/d/dbaas_logs_input_engine.html.markdown (hunk 3 above)
Comment (pdepaepe) [reply website/docs/d/dbaas_logs_input_engine.html.markdown]: Updated, thanks

File: website/docs/d/dbaas_logs_input_engine.html.markdown (hunk 4 above)
Comment (pdepaepe) [reply website/docs/d/dbaas_logs_input_engine.html.markdown]: Updated, thanks

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-check#1207.0 ✔ 
* 
  * checks ✔

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-check#1208.0 ✔ 
* 
  * checks ✔

**Comment by scraly** [comment]
Thanks!

**Pull request merged**